tools/crm_node.c

```diff
@@ -526,6 +526,39 @@ print_quorum(void)
     exit_code = pcmk_rc2exitc(rc);
 }
 
+/*!
+ * \internal
+ * \brief Extend a transaction by removing a node from a CIB section
+ *
+ * \param[in,out] cib        Active CIB connection
+ * \param[in]     element    CIB element containing node name and/or ID
+ * \param[in]     section    CIB section that \p element is in
+ * \param[in]     node_name  Name of node to purge (NULL to leave unspecified)
+ * \param[in]     node_id    Node ID of node to purge (0 to leave unspecified)
+ *
+ * \note At least one of node_name and node_id must be specified.
+ * \return Standard Pacemaker return code
+ */
+static int
+remove_from_section(cib_t *cib, const char *element, const char *section,
+                    const char *node_name, long node_id)
+{
+    xmlNode *xml = NULL;
+    int rc = pcmk_rc_ok;
+
+    xml = create_xml_node(NULL, element);
+    if (xml == NULL) {
+        return pcmk_rc_error;
+    }
+    crm_xml_add(xml, XML_ATTR_UNAME, node_name);
+    if (node_id > 0) {
+        crm_xml_set_id(xml, "%ld", node_id);
+    }
+    rc = cib->cmds->remove(cib, section, xml, cib_transaction);
+    free_xml(xml);
+    return (rc >= 0)? pcmk_rc_ok : pcmk_legacy2rc(rc);
```

Comment:
`rc` is always a legacy return code. It's never a call ID (so never positive), because it's synchronously added to the transaction at client side. `pcmk_legacy2rc(rc)` by itself would be fine unless I'm overlooking something.

Reply:
Yeah, I was going to, but felt uncomfortable assuming it would never (in the future) return a call ID, and I didn't think it hurt. Eventually we'll have something with a standard return code which avoids all such confusion